src/com/amazon/ion/impl/IonTokenConstsX.java

```diff
@@ -513,7 +514,13 @@ public static int decodeSid(CharSequence sidToken)
         assert length > 1;
 
         String digits = sidToken.subSequence(1, length).toString();
-        return Integer.parseInt(digits);
+
+        try {
+            return Integer.parseInt(digits);
+        } catch (Exception e) {
+            throw new UnknownSymbolException(digits);
```

Comment:
The symbol ID isn't necessarily unknown, it just can't fit into an `int`. Therefore, `UnknownSymbolException` isn't the right exception to throw here. Instead, we should just throw an `IonException` and provide `e` as the cause. Something like
```
throw new IonException(String.format("Unable to parse SID %s", digits), e);
```